src/tagging/components/InnerComponents/ValueSelector.js

```diff
@@ -6,51 +6,103 @@ import TaggingPropTypes from '../TaggingPropTypes';
 import customStyles from '../../../forms/select-styles';
 
 class ValueSelector extends React.Component {
-  getvalues = values =>
+  getValues = values =>
     values.map(tag => ({ value: tag.id, label: tag.description }));
 
-  handleChange = (selectedOption) => {
-    this.props.onTagValueChange({
-      description: selectedOption.label,
-      id: selectedOption.value,
-    });
+  /**
+   * Maps the selected options for the select box input.
+   *
+   * @returns {id, label}[]
+   */
+  getActiveValues = () => {
+    const val = [];
+    if (this.props.selectedOption.length > 0) {
+      for (let i = 0; i < this.props.selectedOption.length; i += 1) {
+        val.push({
+          value: this.props.selectedOption[i].id,
+          label: this.props.selectedOption[i].description,
+        });
+      }
+    }
+    return val;
+  };
+
+  /**
+   * Handles the single and multi select value changes.
+   *
+   * Sends an Array of {description, id} to the callback.
+   * If null is received sends and empty array instead
+   * (happens when removing elements one by one).
+   * On reset button (multi select) sends the default empty array.
+   *
+   * @param values {label, value}[] Currently selected values.
+   */
+  handleChange = (values) => {
+    // Deleting all the multi selected items one by one
+    if (values === null) {
+      this.props.onTagValueChange([]);
+      return;
```

Comment:
I would prefer seeing this `return this.props.onTagValueChange([]);` instead of empty return statement (i know it will have the same effect)